var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -226,12 +225,16 @@ def cmake_args(self):
             ])
 
         # CUDA
-        # FIXME  -DDEAL_II_CUDA_FLAGS="-arch=sm_60"
         if '+cuda' in spec:
             options.extend([
                 '-DDEAL_II_WITH_CUDA=ON',
                 '-DDEAL_II_WITH_CXX14=OFF'
             ])
+            cuda_arch = self.variants['cuda_arch'].values
+            if cuda_arch is not None:
+                options.append(
+                    '-DDEAL_II_CUDA_FLAGS=-arch=sm_{0}'.format(cuda_arch)
```

Comment:
@Rombur since you are the driving force behind CUDA in dealii, do is it enough to use these flags or do we need to go with `--generate-code arch=compute_{0},code=sm_{0}`?

Reply:
`-arch=sm_{0}` is sufficient. This is the way I am using. Here is a pretty good explanation of what these flags do https://stackoverflow.com/questions/35656294/cuda-how-to-use-arch-and-code-and-sm-vs-compute Basically, what happens is that nvcc creates an intermediary code (PTX) and this code is recompiled. So you can give an architecture flag for the first "compilation" and another one for the second "compilation". If you use `-arch=sm_` You assume that both flags are the same but this only work if you target one GPU (probably not a problem since you want to compile deal on the machine you will use).

Reply:
> probably not a problem since you want to compile deal on the machine you will use

This would create problems for clusters with heterogeneous cards.  But then wouldn't the flags need to be every combination?  E.g. suppose a rig had one `sm_35` and one `sm_50`.  Then it would need to be

```py
CU_FLAGS = ""
for arch_1 in ("sm_35", "sm_50"):
    for arch_2 in ("sm_35", "sm_50"):
        CU_FLAGS = "{0} --generate-code arch=compute_{1},code=sm_{2}".format(CU_FLAGS, arch_1, arch_2)
```

? Or is that not actually useful?

Reply:
@svenevs 

> wouldn't the flags need to be every combination

I extended `cuda_flags` to do that. But I also don't know whether or not in HPC with different CUDA architectures you need to do that.